solidity/dashboard/src/sagas/liquidity-rewards.js

```diff
@@ -89,6 +99,66 @@ export function* watchFetchLiquidityRewardsData() {
   )
 }
 
+export function* watchLiquidityRewardsEarnedNotification() {
+  for (const pairName of Object.keys(LIQUIDITY_REWARD_PAIRS)) {
+    yield takeLatest(
+      `liquidity_rewards/${pairName}_liquidity_rewards_earned_notification`,
+      processLiquidityRewardEarnedNotification
+    )
+  }
+}
+
+function* processLiquidityRewardEarnedNotification(action) {
+  const displayMessage = action.payload.displayMessage
+
+  const liquidityRewardPairName = action.payload.liquidityRewardPairName
+  const liquidityRewardPair = LIQUIDITY_REWARD_PAIRS[liquidityRewardPairName]
+
+  /** @type LiquidityRewards */
+  const LiquidityRewards = yield getLPRewardsWrapper(liquidityRewardPair)
+  const { liquidityRewards } = yield select()
+  const lastNotificationRewardAmount = new BigNumber(
+    liquidityRewards[liquidityRewardPairName].lastNotificationRewardAmount
+  )
+  const currentReward = yield call(
+    [LiquidityRewards, LiquidityRewards.rewardBalance],
+    action.payload.address
+  )
+  // show the notification if the rewardBalance from LPRewardsContract is greater
+  // than the reward amount that was last time the notification was displayed
+  if (gt(currentReward, lastNotificationRewardAmount)) {
+    if (displayMessage) {
```

Comment:
I think we can ` if (gt(currentReward, lastNotificationRewardAmount) && displayMessage)`